src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAdditionalLinkService.java

```diff
@@ -41,11 +48,20 @@ public AdminCourtAdditionalLinkService(final CourtRepository courtRepository, fi
 
         final List<CourtAdditionalLink> newCourtAdditionalLinks = constructCourtAdditionalLinksEntity(courtEntity, additionalLinks);
         courtAdditionalLinkRepository.deleteCourtAdditionalLinksByCourtId(courtEntity.getId());
-        return courtAdditionalLinkRepository.saveAll(newCourtAdditionalLinks)
+        List<AdditionalLink> newLinks = courtAdditionalLinkRepository.saveAll(newCourtAdditionalLinks)
             .stream()
-            .map(a -> a.getAdditionalLink())
+            .map(CourtAdditionalLink::getAdditionalLink)
             .map(AdditionalLink::new)
             .collect(toList());
+        adminAuditService.saveAudit(
+            AuditType.findByName("Update court additional links"),
+            gson.toJson(courtEntity.getCourtAdditionalLinks()
```

Comment:
This is fine but just some food for thought... could the service do the Json parsing? We'd avoid the need to import and parse json in each service.

Reply:
Some classes send across a non jsonifiable string (just to make it clearer on the admin portal what has happened / ie where postcodes have moved to for example)